src/core/a-entity.js

```diff
@@ -11,6 +11,17 @@ var MULTIPLE_COMPONENT_DELIMITER = '__';
 var OBJECT3D_COMPONENTS = ['position', 'rotation', 'scale', 'visible'];
 var ONCE = {once: true};
 
+var _resolveReadyPromise;
+var isReady = false;
+var isReadyPromise = new Promise(function (resolve) {
+  _resolveReadyPromise = resolve;
+});
+
+function ready () {
+  isReady = true;
+  _resolveReadyPromise();
+}
```

Comment:
Looking at this again, it might make sense for this to live in its own file/module that just governs whether the entire aframe library is ready. I can tidy this up.

This initial PR was only to gather and understand if there's interest/appetite for such a change.

Reply:
sounds good. thanks. In an ideal world we would have a single build that can be loaded via modules (export, import) or script tag. As far as I know that's not possible, right?